### ISSUE TITLE
Parcel won't stop rebuilding in dev mode

### ISSUE THREAD

**Issue by OscarGodson** [parcel-bundler/parcel#3481]
# 🐛 bug report

Running parcel in "dev mode" on OS X just constantly rebuilds which triggers reloads making it impossible to work with :(

## 🎛 Configuration (.babelrc, package.json, cli command)

`package.json`

```js
{
  "name": "ozzie",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "parcel ./index.html",
    "test": "mocha --timeout=10000",
    "dbemulator": "firebase serve --only firestore"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "antd": "^3.20.1",
    "dotenv": "^8.0.0",
    "firebase": "^6.3.0",
    "firebase-init": "^1.3.2",
    "formik": "^1.5.8",
    "moment": "^2.24.0",
    "query-string": "^6.8.2",
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-redux": "^7.1.0",
    "react-redux-firebase": "^3.0.0-alpha.13",
    "react-router-dom": "^5.0.1",
    "recompose": "^0.30.0",
    "redux": "^4.0.4",
    "redux-firestore": "^0.8.0"
  },
  "devDependencies": {
    "@babel/core": "^7.5.5",
    "@babel/plugin-proposal-class-properties": "^7.5.5",
    "@babel/plugin-transform-runtime": "^7.5.5",
    "@babel/preset-env": "^7.5.5",
    "@babel/preset-react": "^7.0.0",
    "@babel/runtime": "^7.5.5",
    "babel-loader": "^8.0.6",
    "babel-polyfill": "^6.26.0",
    "babel-preset-es2015": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "parcel-bundler": "^1.12.3",
    "@firebase/testing": "0.10.0",
    "filesystem": "1.0.1",
    "mocha": "5.2.0",
    "should": "^13.2.3"
  }
}
```

`.babelrc`

```js
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "browsers": [
            ">0.25%",
            "not ie 11",
            "not op_mini all"
          ]
        }
      }
    ],
    "@babel/preset-react"
  ],
  "plugins": [
    "@babel/plugin-transform-runtime",
    "@babel/plugin-proposal-class-properties"
  ]
}
```

## 🤔 Expected Behavior

It should only rebuild and reload when I save a file.

## 😯 Current Behavior

Constantly rebuilding and causing refreshing. The gif below you can see. I'm doing absolutely nothing but sitting and watching my terminal.

![rebuild](https://user-images.githubusercontent.com/20615/64060365-53946f80-cb80-11e9-8f65-c64ef80185db.gif)

## 💁 Possible Solution

Maybe not a solution per se but it'd be nice to get some info on _what_ is causing this refresh. Something is triggering a "change" event, but what?

## 🔦 Context

Just a brain dump but I'm working in IntelliJ. I tried IntelliJ's terminal and iTerm. I've tried deleting `node_modules`, `dist`, `.cache`, and `package-lock`. I've reinstalled npm modules multiple times.

## 💻 Code Sample

I can't publicly share the code but it's a _pretty_ simple React app that uses Firebase and Firestore.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.12.3
| Node             | 10.16.0
| npm/Yarn         | 6.9.0
| Operating System | Mac OS 10.14.1 (18B75)

**Comment by OscarGodson** [comment]
I'm wondering if this has to do with having it inside of a Dropbox directory. I think Dropbox might be creating temp files or touching files when it sees changes. Is there a way to whitelist or blacklist files?

**Comment by DeMoorJasper** [comment]
@OscarGodson issues with the watcher combines with dropbox has been reported in the past probably good to move it out of dropbox

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ccssmnn** [comment]
Had the same issue today on OSX.

I was able to fix it by moving my project into a directory that is not synced with iCloud Drive.

**Comment by liorgrossman** [comment]
Having the same issue here, Parcel continuously restarting whenever Dropbox enters sync mode
Did anyone find a fix or a workaround?

**Comment by DeMoorJasper** [comment]
@liorgrossman it has to do with the watcher detecting file changes made by dropbox... the only workaround is to not have a parcel project in a dropbox folder.

**Comment by bobbwhy** [comment]
I have this issue and I am not using dropbox.
For me, it happened when I switched branches while parcel was running.
Even after switch complete, the issue still happens

**Comment by DeMoorJasper** [comment]
@bobbwhy that sounds like a different issue, feel free to open a new issue

**Comment by ManzDev** [comment]
Same error with a simple html with link to empty CSS/JS files.

parcel: 2.0.0-beta.1 (and 2.0.0-nightly.432 fails too)
node: 14.14.0 with npm 7
so: Debian 10 with WSL

![parcel-error](https://user-images.githubusercontent.com/1507150/97192249-a4612180-179f-11eb-8696-69638a953b45.gif)

**Comment by DeMoorJasper** [comment]
@ManzDev are you running it inside a dropbox or some other kind of cloud storage synced folder?

**Comment by ManzDev** [comment]
> @ManzDev are you running it inside a dropbox or some other kind of cloud storage synced folder?

No, Parcel 1 works fine, but Parcel 2 build constantly. I'm running under WSL (Debian 10). No dropbox.

**Comment by DeMoorJasper** [comment]
@ManzDev strange, can you create a minimal reproduction?

**Comment by ManzDev** [comment]
Hi @DeMoorJasper 

I created this minimal reproduction and works fine (also with node 14.14.0). But in my system, ocurrs infinite rebuilding.

```
FROM node:12.19.0-slim
WORKDIR /app
RUN npm init -y && \
    npm install --save-dev parcel@next
RUN mkdir -p src/css src/js
RUN touch src/css/index.css && touch src/js/index.js
COPY index.html /app/src
CMD ["npx", "parcel", "src/index.html"]
```

**Comment by ManzDev** [comment]
I create a new example from scratch on my system and now works perfectly.

I have been working fine with no problems for a while. 
Sometimes, when write an import to a .js file and save, infinite looping has started.

- If I break (CTLR-C) and restart Parcel, it continues loop.
- If I delete the `.parcel-cache` folder and restart Parcel, works correctly again.

**Comment by DeMoorJasper** [comment]
@ManzDev that definitely sounds like some kind of recursion bug, but kind of need a way to reproduce it to debug it properly.

This probably occurred sometime in the past and you still had a cache entry for it which caused it to happen consistently for you but works fine with a new project. I assume the only thing you can do is if it ever happens again to think what you changed before and try and create a reproduction from based on that. I really wished I could help more here as this does sound like a pretty severe bug.

**Comment by ManzDev** [comment]
I need to test a bit more to see if it happens with some kind of specific pattern. It seems like it doesn't always happens.

I would also like to try on another system. As soon as I know more, I'll post it here. Thank you very much!

**Comment by bp9320** [comment]
I had the same issue on MacOS (Catalina 10.15.7, Node 14.14.0, Parcel 2.0.0-nightly.440). 

Like @ManzDev, deleting the .parcel-cache folder stopped the constant build process (thanks for that tip!), but I noticed it started happening again after I got a build error. I have reproduced it in this way roughly six times, though I have not tried in other projects.

![parcelBug](https://user-images.githubusercontent.com/32176471/98457129-57d7f780-214a-11eb-8171-fb05ca8baef0.gif)

**Comment by AndrewKvalheim** [comment]
This consistently reproduces it for me:

```console
$ yarn add --dev parcel@2.0.0-nightly.465
$ touch a.html
$ echo '<a href="a.html">' > b.html
$ ( sleep 3s && touch a.html ) & parcel watch a.html b.html | grep 'Built in'
✨ Built in 294ms
✨ Built in 21ms
✨ Built in 9ms
✨ Built in 13ms
✨ Built in 9ms
✨ Built in 8ms
✨ Built in 9ms
✨ Built in 12ms
✨ Built in 16ms
✨ Built in 14ms
[…]
```

**Comment by stevematney** [comment]
I was able to get the constant rebuild to stop by following [these instructions for Dropbox to ignore files or folders](https://help.dropbox.com/files-folders/restore-delete/ignored-files), and ignoring `node_modules`, `dist`, and `.parcel-cache`.

I have not gone through the steps to narrow down exactly which folder was the culprit, but I figure it's not very important that Dropbox track those files anyway.

**Comment by mischnic** [comment]
Might or might not be the same bug as https://github.com/parcel-bundler/parcel/issues/5355

**Comment by RohanG0407** [comment]
Every time I make a change in a sass file, it seems to keep infinitely rebuilding. Anyone had this happen to them? I can fix this by deleting parcel-cache and if I make edits in any other files it builds only once. Only sass files it keeps rebuilding

**Comment by DestroyOnLoad** [comment]
> I create a new example from scratch on my system and now works perfectly.
> 
> I have been working fine with no problems for a while.
> Sometimes, when write an import to a .js file and save, infinite looping has started.
> 
> * If I break (CTLR-C) and restart Parcel, it continues loop.
> * If I delete the `.parcel-cache` folder and restart Parcel, works correctly again.

Thank you!  Completely the issue I was dealing with.  Not sure what was in the cache triggering the change event, but it's simple enough to let parcel just rebuild.

**Comment by multimonos** [comment]
i had a similar which was caused by my directory structure.  

Originally I had 
```
.
├── package.json
├── style.css
├── style.css.map
└── style.scss
```
and was running `npx parcel watch ./style.scss`. this was overwriting `/app/style.css` and then re-building in the same location ... which seems to have triggered the infinite loop.

changing folder structure to,
```
.
├── package.json
├── src
│   └── style.scss
├── style.css
└── style.css.map
```
and running `npx parcel watch ./src/style.scss` works and now generates output file at `/app/style.css` as per `"main": "style.css"` in `package.json` without the infinite loop.

**Comment by ayush-ojha-11** [comment]
run this command:  rm -rf .parcel-cache
and then update parcel : npm install --save-dev parcel@latest
and then re build the project.
